homeassistant/components/purpleair/config_flow.py

```diff
@@ -178,6 +181,37 @@ async def async_step_by_coordinates(
 
         return await self.async_step_choose_sensor()
 
+    async def async_step_check_api_key(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the step to check the API key."""
+        if self._reauth_entry:
+            error_step_id = "reauth_confirm"
+        else:
+            error_step_id = "user"
+
+        api_key = self._flow_data[CONF_API_KEY]
+
+        validation = await async_validate_api_key(self.hass, api_key)
+
+        if validation.errors:
+            return self.async_show_form(
+                step_id=error_step_id,
+                data_schema=API_KEY_SCHEMA,
+                errors=validation.errors,
+            )
```

Comment:
Can you refactor this to only show the form for the step in actual step itself?  We aren't supposed to show the form for `user` in the `check_api_key` step